pywbem/_logging.py

```diff
@@ -0,0 +1,402 @@
+        """
+        Setup a single named logger with the characteristics defined on input.
+
+        This function can be used to set up all of the named loggers used by
+        pywbem.
+
+        Parameters:
+          log_compnent (:term:`string`):
+           The name of the logger. It must be one of the
+           names defined in PYWBEM_LOG_COMPONENTS. Subsequent logging users
+           will use this name to reference the parameters of this logger
+
+          log_dest (:term:`string`):
+            String defining the destination for this log. It must be one of the
+            destinations defined in LOG_DESTINATIONS
+
+          log_filename (:term:`string`):
+            Filename to use as logging file if the log destination is `file`.
+            Ignored if log destination is not `file`. The default is
+            DEFAULT_LOG_FILENAME defined in config.py
+
+          log_detail_level (:term:`string`):
+            String defining the level of detail for log output. This is
+            optional.
+            Not all of the loggers use this attribute.  Thus, the ops and http
+            loggers used a defined detail level and always set logging to that
+            level.
+
+          log_level (:term:`string`):
+            String defining the log level. This is optional.  Not all of the
+            loggers use this attribute.  Thus, the ops and http loggers used a
+            defined level and always set logging to that level.
+
+        Return:
+            None
+
+        Exceptions:
+            ValueError - Input cannot be mapped to a defined logger or
+            component.
+            No named logger is created.
+        """
+        if log_dest not in LOG_DESTINATIONS:
+            raise ValueError('Invalid log destination %s. Not in list %s' %
+                             (log_dest, LOG_DESTINATIONS))
+        if log_component not in PYWBEM_LOG_COMPONENTS:
+            raise ValueError('Invalid log component %s. Not in list %s' %
+                             (log_component, PYWBEM_LOG_COMPONENTS))
+        if not log_level:
+            log_level = DEFAULT_LOG_LEVEL
+        if log_level not in LOG_LEVELS:
+            raise ValueError('Invalid log level %s. Not in list %s' %
+                             (log_level, LOG_LEVELS))
+
+        if not log_detail_level:
+            log_detail_level = DEFAULT_LOG_DETAIL_LEVEL
+        if log_detail_level not in LOG_DETAIL_LEVELS:
+            raise ValueError('Invalid log detail %s. Not in list %s' %
+                             (log_detail_level, LOG_DETAIL_LEVELS))
+
+        # If destinations or components is all, recurse for all destinations
+        if log_dest == 'all':
+            for dest in LOG_DESTINATIONS:
+                cls.create_logger(log_component, dest,
+                                  log_filename=log_filename,
+                                  log_detail_level=log_detail_level,
+                                  log_level=log_level)
```

Comment:
The statements between lines 264 and 269 should be removed, because there is no destination "all".